src/main/scala-sbt-0.13/sbt/internal/sbtscalafix/Caching.scala

```diff
@@ -0,0 +1,42 @@
+package sbt.internal.sbtscalafix
+
+import java.io.ByteArrayOutputStream
+
+import sbinary._
+import sbt._
+import scalafix.internal.sbt.Arg
+
+import scala.util.DynamicVariable
+
+object Caching {
+
+  trait CacheKeysStamper
+      extends InputCache[Seq[Arg.CacheKey]]
+      with CacheImplicits {
+    private val output = new DynamicVariable[Output](null)
+
+    implicit val lastModifiedFormat = FileInfo.lastModified.format
+    val baFormat = implicitly[Format[Array[Byte]]]
+    val baEquiv = implicitly[Equiv[Array[Byte]]]
+
+    protected def stamp: Arg.CacheKey => Unit
+
+    final def write[T](t: T)(implicit format: Format[T]): Unit =
+      format.writes(output.value, t)
+
+    override final type Internal = Array[Byte]
+    override final def convert(keys: Seq[Arg.CacheKey]): Array[Byte] = {
+      val baos = new ByteArrayOutputStream()
+      output.withValue(new JavaOutput(baos)) {
+        keys.foreach(stamp)
+      }
+      Hash.apply(baos.toByteArray)
+    }
```

Comment:
this effectively replicates the [1.x behavior](https://github.com/sbt/sbt/commit/b0b9dc5e0f6a13bbba11fbf43b98575e4f73f009) (although with a binary representation and a different hashing algorithm)